### PR TITLE
Remove `origin` from events and assertions

### PR DESCRIPTION
**Pull request by gnieto**
Related to: https://github.com/matrix-org/matrix-spec/pull/998

Setting `origin` field in events/PDUs is not required any more, so I've removed assertions and assignments to this field.

### CODE REVIEW

**Comment by gnieto** [comment]
Apparently, Dendrite is no happy with this change. 
I can just remove the assertions, alternatively, if you are fine with that.

**Comment by DMRobertson** [comment]
> Apparently, Dendrite is no happy with this change.

Probably worth opening a Dendrite bug (if there isn't one for this already).

> I can just remove the assertions, alternatively, if you are fine with that.

Sounds reasonable.

**Comment by richvdh** [comment]
>  I can just remove the assertions, alternatively, if you are fine with that.

If you take this path, please can you comment the places we set `origin` with a link to the dendrite bug?

File: lib/SyTest/Federation/Client.pm
Comment (gnieto) [inline lib/SyTest/Federation/Client.pm:243]: @richvdh Is this similar to what you were expecting? Let me know if you want me to change the message. Thanks!

**Comment by neilalexander** [comment]
matrix-org/dendrite#2737
matrix-org/gomatrixserverlib#341

Please hold off on doing anything until we get these landed.

**Review by neilalexander — APPROVED** [review]
Right, this should be OK now that we've fixed Dendrite. I have reverted aed2b10d9cbfb4ea10ac6d6dcbbf11d276ed7a33 — LGTM if CI is green.

**Review by DMRobertson — APPROVED** [review]
Thanks Neil A for the prompt response!

**Pull request merged**
